chain/network/src/peer/codec.rs

```diff
@@ -78,9 +78,7 @@ impl Decoder for Codec {
             return Ok(None);
         }
 
-        let mut len_bytes: [u8; 4] = [0; 4];
-        len_bytes.copy_from_slice(&buf[0..4]);
-        let len = u32::from_le_bytes(len_bytes);
+        let len = u32::from_le_bytes([buf[0], buf[1], buf[2], buf[3]]);
```

Comment:
Looks like we can use https://docs.rs/bytes/latest/bytes/buf/trait.Buf.html#method.get_u32_le ?

Reply:
@matklad Thanks, that's even better.

Reply:
Meta note on how to know such things: 

I've noticed that both versions do not exactly spark joy. At the same time, reading length prefix seems like a common operation one might do, and it feels like there *should* be a better way of doing this. I wanted to grep some tokio code to see if I can find the relevant pattern. But before that, I reasoned that it might be a good idea to check the docs for the type (it seems reasonable that the functionality can live on the `BytesMut` type itself). I actually was looking for something with `&mut BytesMut -> Option<[u8; N]>` signature, but just stumbled upon the very specific API.

Reply:
@matklad I tried looking for some apis, that convert `&[u8]` to `[u8;4]`, though I couldn't find anything. 
`fn get_u32_le(&mut self) -> u32` doesn't even have `[u8;4]` in it's signature either.

Though, I should have checked the `buf` structs. Anyway, it was a good idea to check.